clients/src/main/java/org/apache/kafka/common/requests/OffsetFetchResponse.java

```diff
@@ -146,6 +84,32 @@ public PartitionData(long offset,
         public boolean hasError() {
             return this.error != Errors.NONE;
         }
+
+        @Override
+        public boolean equals(Object other) {
+            if (!(other instanceof PartitionData))
+                return false;
+            PartitionData otherPartition = (PartitionData) other;
+            return this.offset == otherPartition.offset
+                       && this.leaderEpoch.equals(otherPartition.leaderEpoch)
+                       && this.metadata.equals(otherPartition.metadata)
```

Comment:
Metadata is nullable. We should make sure to have some test cases which include null metadata.